processor/processor.go

```diff
@@ -2230,22 +2279,92 @@ func (proc *HandleT) mainPipeline(ctx context.Context) {
 		}
 	}()
 
-	chStore := make(chan storeMessage, bufferSize)
+	triggerStore := make(chan int, 1)
+	chStore := make(chan storeMessage, subJobCount)
 	wg.Add(1)
+
 	go func() {
 		defer wg.Done()
 		defer close(chStore)
-
+		defer close(triggerStore)
 		for msg := range chTrans {
+
 			chStore <- proc.transformations(msg)
+			//since we don't want to call DB write for each sub-jobs. That's why we are waiting until all the sub-jobs are processed.
+			//Once all the sub-jobs are processed, we trigger the DB write by `triggerStore` channel.
+			if len(chStore) == subJobCount || !msg.isSplit {
+				triggerStore <- 1
```

Comment:
In case you follow the configurable `subJobSize` strategy, then you could push here the calculated `subJobCount`, instead of the literal value `1`

Reply:
we are capturing that in the `msg.subJobCount`. I have changed `<-1` to `<-true`.

Reply:
the reason I have included `subJobCount` in the struct, is to maintain consistency across all the the go routine in the mainpipeline